addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionScanRule.java

```diff
@@ -651,446 +651,627 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
             countUnionBasedRequests = 0;
             countOrderByBasedRequests = 0;
 
-            // Check 1: Check for Error Based SQL Injection (actual error messages).
-            // for each SQL metacharacter combination to try
-            for (int sqlErrorStringIndex = 0;
-                    sqlErrorStringIndex < SQL_CHECK_ERR.length
-                            && !sqlInjectionFoundForUrl
-                            && doSpecificErrorBased
-                            && countErrorBasedRequests < doErrorMaxRequests;
-                    sqlErrorStringIndex++) {
-
-                // work through the attack using each of the following strings as a prefix: the
-                // empty string, and the original value
-                // Note: this doubles the amount of work done by the scanner, but is necessary in
-                // some cases
-                String[] prefixStrings;
-                if (origParamValue != null) {
-                    prefixStrings = new String[] {"", origParamValue};
-                } else {
-                    prefixStrings = new String[] {""};
+            testErrorBasedSqlInjection(param, origParamValue);
```

Comment:
I think this is changing the behaviour, the returns that would return from the scan method are now just returning from the extracted methods (and flow continues here), I didn't check the whole code yet but at least it should have isStop checks between them.

Reply:
Good catch. Updated to skip remaining cases if either `isStop()` of `sqlInjectionFoundForUrl`